packages/yarnpkg-fslib/sources/patchFs/patchFs.ts

```diff
@@ -136,6 +138,16 @@ type ReadArgumentsOptions = [
 type ReadArgumentsCallback = [fd: number, callback: ReadCallback];
 //#endregion
 
+export function applyFsLayer(origFs: typeof fs, factory: (baseFs: FakeFS<PortablePath>) => FakeFS<PortablePath>) {
+  // We must copy the fs into a local, because otherwise
+  // 1. we would make the NodeFS instance use the function that we patched (infinite loop)
+  // 2. Object.create(fs) isn't enough, since it won't prevent the proto from being modified
+  const localFs = {...origFs};
+  const nodeFs = new NodeFS(localFs);
+
+  patchFs(fs, new PosixFS(factory(nodeFs)));
+}
```

Comment:
I wonder if we could find a way to avoid repeating those indirections when applying multiple mount layers ... perhaps by allowing `MountFS` to handle an array of mount configuration instead of a single one?